modules/p2/ting_openlist/include/ting_openlist_analysis.inc

```diff
@@ -0,0 +1,150 @@
+<?php
+
+/**
+ * @file
+ * Functions for analysis of local OpenList Entities
+ */
+
+/**
+ * Get providers.
+ *
+ * @return int
+ *   Number of providers.
+ */
+function ting_openlist_local_analysis_data_providers() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Get providers with an openlist uid.
+ *
+ * @return int
+ *   Number of providers.
+ */
+function ting_openlist_local_analysis_data_providers_id() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  $query->condition('openlist_uid', NULL, 'IS NOT');
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Get providers which is synced.
+ *
+ * @return int
+ *   Number of synced users.
+ */
+function ting_openlist_local_analysis_data_providers_openlist_is_sync() {
+  $query = db_select('users', 'u');
+  $query->join('users_roles', 'ur', 'ur.uid = u.uid');
+  $query->join('role', 'r', "r.rid = ur.rid AND r.name = 'provider'");
+  $query->condition('openlist_modified', 0, '>');
+  return $query->countQuery()->execute()->fetchField();
+}
+
+/**
+ * Count the types of lists.
+ *
+ * @return array
+ *   An associative array, where the key is the list_type_X (X being number of
+ *   lists) and the key is number of occurences.
```

Comment:
This sounds odd to me.

As I read it if there are 42 lists of type `books_read` then the array would contain the following key-value pair:

`books_read_42 => 42`

Perhaps I do not understand the inner workings of the OpenList analysis or I misunderstand the comment.